base/pkg/dir.jl

```diff
@@ -35,15 +35,20 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
     end
     dir = path()
     info("Initializing package repository $dir")
+    temp_dir = joinpath(dir, "TEMP")
     metadata_dir = joinpath(dir, "METADATA")
+    #Check if TEMP dir exists from previous failed init
+    if isdir(temp_dir)
```

Comment:
it's also best not to call `rm` when you can avoid it, since the user would be most unhappy to lose a working directory they happened to call `TEMP`.